ngraph/test/backend/proposal.in.cpp

```diff
@@ -0,0 +1,2619 @@
+        0.0000000f, 117.1594696f, 118.1435699f, 168.0880280f, 178.6725311f, // 85
+        0.0000000f, 0.0000000f,   150.1572571f, 271.8343201f, 209.0000000f, // 90
+        0.0000000f, 206.6396332f, 76.6708832f,  270.4295349f, 151.5276794f, // 95
+        0.0000000f, 91.6318054f,  169.5779114f, 179.0037231f, 209.0000000f, // 100
+        0.0000000f, 0.0000000f,   0.0000000f,   43.0686874f,  36.4258842f,  // 105
+        0.0000000f, 0.0000000f,   0.0000000f,   105.7007523f, 209.0000000f, // 110
+        0.0000000f, 43.1367035f,  0.0000000f,   275.3395996f, 188.5202179f, // 115
+        0.0000000f, 110.2392731f, 68.0856171f,  164.6310883f, 146.0089111f, // 120
+    };
+
+    size_t proposal_ref_size = sizeof(proposal_ref) / sizeof(proposal_ref[0]);
+
+    const float iou_threshold = 0.7f;
+    const int min_bbox_size = 16;
+    const int feature_stride = 16;
+    const int pre_nms_topn = 6000;
+    const int post_nms_topn = 25;
+    const int image_w = 350;
+    const int image_h = 210;
+    const int image_z = 1;
+    const std::vector<float> ratios = {0.5f, 1.0f, 2.0f};
+    const std::vector<float> scales = {8.0f, 16.0f, 32.0f};
+
+    Shape class_probs_shape{1, 18, 23, 14};
+    Shape bbox_deltas_shape{1, 36, 23, 14};
+    Shape image_shape_shape{3};
+    Shape output_shape = {25, 5};
+
+    op::ProposalAttrs attrs;
+    attrs.base_size = min_bbox_size;
+    attrs.min_size = min_bbox_size;
+    attrs.pre_nms_topn = pre_nms_topn;
+    attrs.post_nms_topn = post_nms_topn;
+    attrs.nms_thresh = iou_threshold;
+    attrs.feat_stride = feature_stride;
+    attrs.min_size = min_bbox_size;
+    attrs.ratio = ratios;
+    attrs.scale = scales;
+    attrs.clip_before_nms = true;
+    attrs.clip_after_nms = false;
+    attrs.normalize = false;
+    attrs.box_size_scale = 1.0f;
+    attrs.box_coordinate_scale = 1.0f;
+    attrs.framework = "";
+    attrs.infer_probs = false;
+
+    auto class_probs_param = make_shared<op::Parameter>(element::f32, class_probs_shape);
+    auto bbox_deltas_param = make_shared<op::Parameter>(element::f32, bbox_deltas_shape);
+    auto image_shape_param = make_shared<op::Parameter>(element::f32, image_shape_shape);
+
+    auto proposal = make_shared<op::v0::Proposal>(
+        class_probs_param, bbox_deltas_param, image_shape_param, attrs);
+    auto f = make_shared<Function>(
+        proposal, ParameterVector{class_probs_param, bbox_deltas_param, image_shape_param});
+
+    std::vector<float> c(&cls_scores_data[0], &cls_scores_data[cls_scores_data_size]);
+    std::vector<float> b(&bbox_pred_data[0], &bbox_pred_data[bbox_pred_data_size]);
+    std::vector<float> i{image_h, image_w, image_z};
+
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_input<float>(class_probs_shape, c);
+    test_case.add_input<float>(bbox_deltas_shape, b);
+    test_case.add_input<float>(image_shape_shape, i);
+
+    std::vector<float> o(&proposal_ref[0], &proposal_ref[proposal_ref_size]);
```

Comment:
Consider https://en.cppreference.com/w/cpp/iterator/begin second overloading.
`std::begin` and `std::end` is more generic and 
```
using std::begin;
using std::end;
vectro v{begin(in), end(in)};
```
is the most (->https://en.cppreference.com/w/cpp/language/adl ).